utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,296 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    /// Increments the digits by 1. if the digits are empty, it will add
+    /// an element with value 1. If there are some trailing zeros,
+    /// it will be reomved from `self.digits`.
+    fn increment_abs_by_one(&mut self) -> Result<(), Error> {
+        for (zero_count, digit) in self.digits.iter_mut().rev().enumerate() {
+            *digit += 1;
+            if *digit < 10 {
+                self.digits.truncate(self.digits.len() - zero_count);
+                #[cfg(debug_assertions)]
+                self.check_invariants();
+                return Ok(());
+            }
+        }
+
+        self.digits.clear();
+
+        if self.magnitude == i16::MAX {
+            self.magnitude = 0;
+
+            #[cfg(debug_assertions)]
+            self.check_invariants();
+            return Err(Error::Limit);
+        }
+
+        // Still a carry, carry one to the next magnitude.
+        self.digits.push(1);
+        self.magnitude += 1;
+
+        if self.upper_magnitude < self.magnitude {
+            self.upper_magnitude = self.magnitude;
+        }
+
+        #[cfg(debug_assertions)]
+        self.check_invariants();
+        Ok(())
+    }
+
+    /// Removes the trailing zeros in `self.digits`
+    fn remove_trailing_zeros_from_digits_list(&mut self) {
+        let no_of_trailing_zeros = self
+            .digits
+            .iter()
+            .rev()
+            .take_while(|&digit| *digit == 0)
+            .count();
+
+        self.digits.truncate(crate::ops::i16_sub_unsigned(
+            self.digits.len() as i16,
+            no_of_trailing_zeros as u16,
+        ) as usize);
```

Comment:
I just noticed that this is not quite right. `self.digits.len()` is safe to cast to a `u16` but not a `i16`.

You can just write:

```rust
self.digits.truncate(self.digits.len() - no_of_trailing_zeros)
```

Reply:
Fixed in #1981